From e248ac7257a43099b6df9c8c40b582d6815aa83a Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 21 Oct 2011 20:00:07 +0800
Subject: [PATCH 162/280] ENGR00160518 vpu: Workaround to disable btp to use
 iram for vc1

There is one ticket:
ticket #179: vpu is blocked in mpeg2 DecGetInitialInfo after VC1 stream

Workaround to disablt BTP flag in iram usage can pass the decoder. Will
remove the workaround after the root cause in f/w is found.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.h  | 2 +-
 vpu/vpu_util.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 96ce7ea..d6147c4 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -639,7 +639,7 @@ typedef struct vpu_versioninfo {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 3, 3)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 3, 4)
 
 extern unsigned int system_rev;
 
diff --git a/vpu/vpu_util.c b/vpu/vpu_util.c
index a8d871e..4729e96 100644
--- a/vpu/vpu_util.c
+++ b/vpu/vpu_util.c
@@ -980,7 +980,7 @@ out:
 		psecAxiIramInfo->useIpEnable = psecAxiIramInfo->useHostIpEnable;
 		psecAxiIramInfo->useDbkEnable = psecAxiIramInfo->useHostDbkEnable;
 		psecAxiIramInfo->useOvlEnable = psecAxiIramInfo->useHostOvlEnable;
-		psecAxiIramInfo->useBtpEnable = psecAxiIramInfo->useHostBtpEnable;
+		psecAxiIramInfo->useBtpEnable = psecAxiIramInfo->useHostBtpEnable = 0;
 	}
 
 	if (((parm->codecMode == VC1_DEC) && !psecAxiIramInfo->useHostOvlEnable) ||
-- 
1.8.0

